### PR TITLE
fix(datepicker): make date range aria-live and fix active date logic

### LINKED ISSUE #11128: Datepicker month/year selection not announced w/ screen-reader
### Summary
Choose month & year range 2016 - 2039 is not being announced in calendar available under basic calendar

### Reproduction on material.angular.io
1. Linear navigate to ' Components ' menu item and click 'Enter' to open components page.                                                                                                                                          
2. Linear navigate forward and set focus on expand collapse left hand menu item ' Form Controls'                                                                                                                               
3. Click 'Enter' key to expand OR collapse this menu item                                                                                                                                                                                                                 
4. Linear navigate into "Datepicker" and click ENTER keystroke.                                                                            
5. Navigate into tabs "Overview" and Click ENTER                                                                                                                                                                                                                             
6. Navigate into "Basic datepicker" heading and navigate into calendar button and press enter button.                                                                                                                
7. Navigate into 'Choose month & year" and click enter button

### Additional comments
Observed on other datepicker components as well.

### LINKED ISSUE #11129: Datepicker not communicating year status on month/year view
### Summary
Upon click on 'Choose month & year' , status of years displayed is not mentioned. no earcon as well.

### Reproduction on material.angular.io
1. Linear navigate to ' Components ' menu item and click 'Enter' to open components page.                                                                                                                                          
2. Linear navigate forward and set focus on expand collapse left hand menu item ' Form Controls'                                                                                                                               
3. Click 'Enter' key to expand OR collapse this menu item                                                                                                                                                                                                                 
4. Linear navigate into "Datepicker" and click ENTER keystroke.                                                                            
5. Navigate into tabs "Overview" and Click ENTER                                                                                                                                                                                                                             
6. Navigate into "Basic datepicker" heading and navigate into calendar button and press enter button.                                                                                                                
7. Navigate into 'Choose month & year" and click enter button

### Additional comments
Observed on other datepicker components as well.

### LINKED ISSUE #11132: Datepicker not announcing status when navigating year blocks
### Summary
Status OR earecon is not being announced upon click on 'previous 20 years' OR 'Next 20 years' button within calendar 

### Reproduction on material.angular.io
1. Linear navigate to ' Components ' menu item and click 'Enter' to open components page.                                                                                                                                          
2. Linear navigate forward and set focus on expand collapse left hand menu item ' Form Controls'                                                                                                                               
3. Click 'Enter' key to expand OR collapse this menu item                                                                                                                                                                                                                 
4. Linear navigate into "Datepicker" and click ENTER keystroke.                                                                            
5. Navigate into tabs "Overview" and Click ENTER                                                                                                                                                                                                                             
6. Navigate into "Basic datepicker" heading and navigate into calendar button and press enter button.                                                                                                                
7. Navigate into  'previous 20 years' OR 'Next 20 years' button's and click enter button to observe for status or earcon announcement.

### Additional comments
Observed on other datepicker components as well.

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #11128 
fixes #11129 
fixes #11132

### CODE REVIEW

**Comment by jelbourn** [comment]
Which screen-readers did you try this in?

**Comment by mmalerba** [comment]
Only on ChromeVox. Now that you mentioned it I tried VoiceOver and it doesn't seem to respect the `aria-live=polite`. Any ideas? Should I do `assertive` instead?

**Comment by jelbourn** [comment]
Try using the `LiveAnnouncer`? The reason we have that is because there are reportedly issues with using more than one `aria-live` region on some environments

**Comment by mmalerba** [comment]
marking this as `target: minor` since it will wind up depending on features only available in the next minor

**Comment by mmalerba** [comment]
@jelbourn switched to `cdkAriaLive` and tested on ChromeVox, JAWS, NVDA, VoiceOver - seems to work on all of them now.

File: src/lib/datepicker/calendar-header.html [hunk 1]
```diff
@@ -1,9 +1,11 @@
 <div class="mat-calendar-header">
   <div class="mat-calendar-controls">
     <button mat-button type="button" class="mat-calendar-period-button"
-            (click)="currentPeriodClicked()" [attr.aria-label]="periodButtonLabel">
+            (click)="currentPeriodClicked()" [attr.aria-label]="periodButtonLabel"
+            cdkAriaLive="polite">
```
Comment (jelbourn) [inline src/lib/datepicker/calendar-header.html]: Just noticed this now, but we should make `cdkAriaLive` default to `polite` in a separate PR

File: src/lib/datepicker/multi-year-view.ts [hunk 2]
```diff
@@ -101,6 +101,9 @@ export class MatMultiYearView<D> implements AfterContentInit {
   /** Emits the selected year. This doesn't imply a change on the selected date */
   @Output() readonly yearSelected: EventEmitter<D> = new EventEmitter<D>();
 
+  /** Emits when any date is activated. */
+  @Output() readonly activeDateChange: EventEmitter<D> = new EventEmitter<D>();
```
Comment (jelbourn) [inline src/lib/datepicker/multi-year-view.ts]: Add unit test for these new outputs?

File: src/lib/datepicker/calendar-header.html (hunk 1 above)
Comment (mmalerba) [reply src/lib/datepicker/calendar-header.html]: created #11618 so we don't forget

File: src/lib/datepicker/multi-year-view.ts (hunk 2 above)
Comment (mmalerba) [reply src/lib/datepicker/multi-year-view.ts]: Updated the tests to work the same way as month-view tests, which test the `@Output()`

**Comment by mmalerba** [comment]
@jelbourn PTAL

**Pull request merged**